### DISCUSSION TITLE
Interpolate properties in the  model

### DISCUSSION THREAD

**Discussion by MarinCronograph** [gempy-project/gempy#953] [General]
Hi,

I managed to get structural model , shown in the  attachment.

But what i am really interested is, can I incorporate well data in a grid and then distribute it  to gain 3D property of for example porosity.

I guess that would go having XYZ data plus Porosity in a Data Frame then using SCIPY to grid the data. Probably this could go. 

But could it be visualized, the porosity model?

b rgds

![field](https://github.com/user-attachments/assets/4045586f-c6c3-4b94-851b-e36891a7517f)

**Comment by javoha** [comment]
Hi, 
so this depends a lot on how advanced you want your property distribution to be. 
You can get the regular grid and the corresponding lithology IDs by using:
`geo_model.grid.regular_grid.values`
`geo_model.solutions.raw_arrays.lith_block`

If you just want a single property per structural element (e.g let's say yellow rocke has a porosity of 0.1, red rock has a porosity of 0.05, etc) you can just create a copy of the lithology ID array and populate it with these values accordingly.
If you want a property distribution (e.g a random field or a kriging interpolation or something like that) you can extract the grid coordinates per structural element and perform your choosen operation (using input data from boreholes or whatever if you like) as input data. In case of kriging based methods you would however need a reasonable variogram model.
Visualization can be done in many ways. Gempys 3D visualization is based on Pyvista, which you can adapt for 3D visualization of property blocks  (but there are of course many other options).
Here is an example of a gaussian field on a single element of a gempy model (example 7):
![image](https://github.com/user-attachments/assets/1a6d96f6-9b7e-47e3-a075-0245ef1199cf)
Is this what you are looking for?

Cheers,
Jan
